### DISCUSSION TITLE
Help picking correct Coral Accelerator

### DISCUSSION THREAD

**Discussion by adamburgoyne** [blakeblackshear/frigate#1750] [Ask A Question]
Hey, I'm finding Frigate pretty good during my test runs with CPU detection so next step I guess is to move to a Coral module to reduce that heavy CPU usage. I'm not very good when it comes to the hardware side of things but have a Lenovo Tiny M900 and I'm not too sure which Coral to get.... I have a feeling I might be able to get a "M.2 Accelerator with Dual Edge TPU" as I think the Mini has a spare M2 2230 port as I'm not using any WiFi modules. 

I've attached a link and picture of the board (I think the M2 port is bottom left labelled "WiFi") if anyone can advise if this is correct? Many thanks!

https://uk.rs-online.com/web/p/microcontroller-development-tools/2133255

![20210909_103306](https://user-images.githubusercontent.com/1240218/132663744-5b312588-d3f7-4778-8297-8ca61d00a452.jpeg)
![20210909_103310](https://user-images.githubusercontent.com/1240218/132663757-7e94db74-0f27-45c3-a137-e02f01a8f1ed.jpeg)

**Comment by lukedukeus** [comment]
That port looks like mSata, not M.2. [This](https://coral.ai/products/pcie-accelerator) looks like it would work. If you are running HA through proxmox or esxi, I would recomend using mSata, otherwise USB, as you can use it with other machines

**Reply by adamburgoyne** [reply]
I am indeed running HA through proxmox but the USB devices are seemingly rare (and expensive) to get in the UK at the moment, I'd prefer something in-case if possible too. It could be mSata, but looking at the link you sent it doesn't look like there's enough contacts on the one side of the chip compared to the M.2 one linked. 

I did eventually find an article on the M900 that states the below which does match the M.2 coral size (2230) and also on the board you can see that screw hole below the port which appears to line up with the one on the M.2 Coral chip. So I think I have an M.2 port!

_"There is also a M.2 2230 slot that is usually reserved for WiFi. On these, you will likely find a Qualcomm Atheros or an Intel 802.11ac WiFi + Bluetooth unit if they are installed."_

**Reply by lukedukeus** [reply]
😳 My bad, that is a M.2 slot. Just make sure that it has the same keying (B/M) as the coral you buy.

**Reply by tve** [reply]
FYI, supposedly mouser is getting a whole slew in October: https://www.mouser.co.uk/ProductDetail/Coral/G950-06809-01?qs=u16ybLDytRbcxxqFKdbhgQ%3D%3D
